mongo/errors.go

```diff
@@ -162,28 +161,34 @@ func writeErrorsFromDriverWriteErrors(errs driver.WriteErrors) WriteErrors {
 	return wes
 }
 
-// WriteConcernError is a write concern failure that occurred as a result of a
-// write operation.
+// WriteConcernError represents a write concern failure during execution of a write operation. This error type is only
+// returned as part of a WriteException or a BulkWriteException.
 type WriteConcernError struct {
 	Name    string
 	Code    int
 	Message string
-	Details bson.Raw
+	Details bson.Raw // A document identifying the write concern setting related to the error.
 }
 
+// Error implements the error interface.
 func (wce WriteConcernError) Error() string {
 	if wce.Name != "" {
 		return fmt.Sprintf("(%v) %v", wce.Name, wce.Message)
 	}
 	return wce.Message
 }
 
-// WriteException is an error for a non-bulk write operation.
+// WriteException is the error type returned by the InsertOne, DeleteOne, DeleteMany, UpdateOne, UpdateMany, and
```

Comment:
Just confirming this is true, since it's a bit odd for a single-write operation to return a list of WriteError instead of just a single one.

Reply:
This is true.